bin/add-prettier-scripts.js

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env node
+
+const fs = require('fs');
+const path = require('path');
+const packageJson = require('../package.json');
+
+const { name } = packageJson;
+
+/**
+ * Adds Prettier scripts to the package.json file
+ */
+const addPrettierScripts = () => {
+  console.info('Adding Prettier scripts to package.json...');
+
+  const packageJsonPath = path.join(process.cwd(), 'package.json');
+  if (!fs.existsSync(packageJsonPath)) {
+    console.error('Error: package.json not found in the current directory');
+    process.exit(1);
+  }
+
+  try {
+    const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'));
+
+    packageJson.scripts = {
+      ...packageJson.scripts,
+      'prettier:check': 'prettier --check "**/*.{ts,tsx,js,jsx}" "**/*.json" "**/*.md"',
+      'prettier:write': 'prettier --write "**/*.{ts,tsx,js,jsx}" "**/*.json" "**/*.md"',
+    };
+
+    fs.writeFileSync(packageJsonPath, `${JSON.stringify(packageJson, null, 2)}\n`);
+
+     const prettierConfigPath = path.join(process.cwd(), '.prettierrc.js');
+    if (!fs.existsSync(prettierConfigPath)) {
+      console.info('Creating .prettierrc.js file...');
+      const prettierConfig =
+`const sharedConfig = require('${name}/.prettierrc.json');
+
+module.exports = sharedConfig;`;
+      fs.writeFileSync(prettierConfigPath, prettierConfig);
+    }
+    const prettierIgnorePath = path.join(process.cwd(), '.prettierignore');
+    if (!fs.existsSync(prettierIgnorePath)) {
+      console.info('Creating .prettierignore file...');
+      const ignoreContent =
+`node_modules
+coverage
+lib
+tsdocs
+npm-shrinkwrap.json
+package-lock.json
+.eslintrc
+*.yml
```

Comment:
Can we override yaml file rules in .prettierrc.json instead of completely opting out? Formatting on save and handling extra spaces, newlines etc is nice if we can get it for free imo.

e.g. if we just want to prefer double quote

```
"overrides": [
    {
      "files": ["*.yml", "*.yaml"],
      "options": {
        "singleQuote": false
      }
```

Reply:
Nice, I'll do that!